packages/xo-server-backup-reports/src/index.js

```diff
@@ -146,41 +260,162 @@ class BackupReportsXoPlugin {
   }
 
   test({ runId }) {
-    return this._backupNgListener(undefined, undefined, undefined, runId)
+    return this._wrapper(undefined, undefined, undefined, runId, true)
   }
 
   unload() {
     this._xo.removeListener('job:terminated', this._report)
   }
 
-  _wrapper(status, job, schedule, runJobId) {
-    if (job.type === 'metadataBackup') {
-      return
+  async _wrapper(status, job, schedule, runJobId, test) {
+    const xo = this._xo
+
+    try {
+      if (test || job.type === 'backup' || job.type === 'metadataBackup') {
+        const log = await xo.getBackupNgLogs(runJobId)
+        if (log === undefined) {
+          throw new Error(`no log found with runId=${JSON.stringify(runJobId)}`)
+        }
+
+        const reportWhen = log.data.reportWhen
+        if (
+          reportWhen === 'never' ||
+          (reportWhen === 'failure' && log.status === 'success')
+        ) {
+          return
+        }
```

Comment:
It should probably be bypassed if a report has been explicitly requested.

Reply:
No answer on this?

Reply:
I thought i did the [required changes](https://github.com/vatesfr/xen-orchestra/pull/4084/files#diff-fc20916414605cc911d77f3aad7bf07cR443)

Reply:
Did explicitly requested means clicking on the plugin test button?

Reply:
Take care of the other comment, we'll go back on this one after.